teps/0001-tekton-enhancement-proposal-process.md

```diff
@@ -0,0 +1,369 @@
+adopters need a forward looking road map in order to plan their
+adoption strategy.
+
+Before this proposal, there is not a standard way or template
+to create project enhancements. We rely on
+documents hosted on Google docs, without a standard template explaining
+the change. Once a proposal is done, via a design docs, it tends
+to be hard to follow what happens with the proposal: updates on the proposal in
+reaction to comments, state of the proposal (when is it accepted, or
+rejected).
+
+The purpose of the TEP process is to reduce the amount of "tribal
+knowledge" in our community. By moving decisions from a smattering of
+mailing lists, video calls and hallway conversations into a well
+tracked artifact, this process aims to enhance communication and
+discoverability.
+
+A TEP is broken into sections which can be merged into source control
+incrementally in order to support an iterative development process. An
+important goal of the TEP process is ensuring that the process for
+submitting the content contained in [design proposals][] is both clear
+and efficient. The TEP process is intended to create high quality
+uniform design and implementation documents for WGs to deliberate.
+
+[road to Go 2]: https://blog.golang.org/toward-go2) 
+[design proposals]: https://github.com/kubernetes/community/tree/master/contributors/design-proposals
+
+## Stewardship
+The following DACI model indentifies the responsible parties for TEPs:
+
+**Workstream** | **Driver** | **Approver** | **Contributor** |
+**Informed** --- | --- | --- | --- | ---
+| TEP Process Stewardship | TG members | TG members | TG members | Community |
+| Enhancement delivery | Enhancement Owner | SIG leadership (SIG Chairs + TLs) | Enhancement Implementer(s) (may overlap with Driver) | Community |
+
+*TG members: Tekton Governing board members*
+
+## Reference-level explanation
+
+### What type of work should be tracked by a TEP
+
+The definition of what constitutes an "enhancement" is a foundational
+concern for the Tekton project. Roughly any Tekton user or
+operator facing enhancement should follow the TEP process. If an
+enhancement would be described in either written or verbal
+communication to anyone besides the TEP author or developer, then
+consider creating a TEP.
+
+Similarly, any technical effort (refactoring, major architectural
+change) that will impact a large section of the development community
+should also be communicated widely. The TEP process is suited for this
+even if it will have zero impact on the typical user or operator.
+
+As the local bodies of governance, WGs should have broad latitude in
+describing what constitutes an enhancement which should be tracked
+through the TEP process. WGs may find it helpful to enumerate what
+_does not_ require a TEP rather than what does. WGs also have the
+freedom to customize the TEP template according to their WG specific
+concerns. For example, the TEP template used to track API changes will
```

Comment:
We still might want to include a list of sections that *should* be adopted by all WGs